examples/arkode/C_serial/ark_conserved_exp_entropy_ark.c

```diff
@@ -0,0 +1,473 @@
+    printf("   Total linear solver setups = %li\n", nsetups);
+    printf("   Total number of Jacobian evaluations = %li\n", nje);
+    printf("   Total RHS evals for setting up the linear system = %li\n", nfeLS);
+  }
+
+  if (relax)
+  {
+    flag = ARKStepGetNumRelaxFails(arkode_mem, &nrf);
+    check_flag(flag, "ARKStepGetNumRelaxFails");
+
+    flag = ARKStepGetNumRelaxFnEvals(arkode_mem, &nre);
+    check_flag(flag, "ARKStepGetNumRelaxFnEvals");
+
+    flag = ARKStepGetNumRelaxJacEvals(arkode_mem, &nrje);
+    check_flag(flag, "ARKStepGetNumRelaxJacEvals");
+
+    flag = ARKStepGetNumRelaxSolveIters(arkode_mem, &nrnlsi);
+    check_flag(flag, "ARKStepGetNumRelaxSolveIters");
+
+    flag = ARKStepGetNumRelaxSolveFails(arkode_mem, &nrnlsf);
+    check_flag(flag, "ARKStepGetNumRelaxSolveFails");
+
+    printf("   Total Relaxation fails     = %li\n", nrf);
+    printf("   Total Relaxation Fn evals  = %li\n", nre);
+    printf("   Total Relaxation Jac evals = %li\n", nrje);
+    printf("   Total Relaxation NLS iters = %li\n", nrnlsi);
+    printf("   Total Relaxation NLS fails = %li\n", nrnlsf);
+  }
+  printf("\n");
+
+  /* -------- *
+   * Clean up *
+   * -------- */
+
+  /* Free ARKStep integrator and SUNDIALS objects */
+  ARKStepFree(&arkode_mem);
+  SUNLinSolFree(LS);
+  SUNMatDestroy(A);
+  N_VDestroy(y);
+  N_VDestroy(ytrue);
+  SUNContext_Free(&ctx);
+
+  return flag;
+}
+
+/* ----------------------- *
+ * User-supplied functions *
+ * ----------------------- */
+
+/* ODE RHS function f(t,y). */
+int f(sunrealtype t, N_Vector y, N_Vector ydot, void* user_data)
+{
+  sunrealtype* ydata = N_VGetArrayPointer(y);
+  sunrealtype* fdata = N_VGetArrayPointer(ydot);
+
+  fdata[0] = -EXP(ydata[1]);
+  fdata[1] = EXP(ydata[0]);
+
+  return 0;
+}
+
+/* ODE RHS Jacobian function J(t,y) = df/dy. */
+int Jac(sunrealtype t, N_Vector y, N_Vector fy, SUNMatrix J, void* user_data,
+        N_Vector tmp1, N_Vector tmp2, N_Vector tmp3)
+{
+  sunrealtype* ydata = N_VGetArrayPointer(y);
+  sunrealtype* Jdata = SUNDenseMatrix_Data(J);
+
+  /* column 0 */
+  Jdata[0] = SUN_RCONST(0.0);
+  Jdata[1] = EXP(ydata[0]);
+
+  /* column 1 */
+  Jdata[2] = -EXP(ydata[1]);
+  Jdata[3] = SUN_RCONST(0.0);
+
+  return 0;
+}
+
+/* Entropy function e(y) */
+int Ent(N_Vector y, sunrealtype* e, void* user_data)
+{
+  sunrealtype* ydata = N_VGetArrayPointer(y);
+
+  *e = EXP(ydata[0]) + EXP(ydata[1]);
+
+  return 0;
+}
+
+/* Entropy function Jacobian Je(y) = de/dy */
+int JacEnt(N_Vector y, N_Vector J, void* user_data)
+{
+  sunrealtype* ydata = N_VGetArrayPointer(y);
+  sunrealtype* jdata = N_VGetArrayPointer(J);
+
+  jdata[0] = EXP(ydata[0]);
+  jdata[1] = EXP(ydata[1]);
+
+  return 0;
+}
+
+/* ----------------- *
+ * Utility functions *
+ * ----------------- */
+
+/* Analytic solution */
+int ans(sunrealtype t, N_Vector y)
+{
+  sunrealtype a, b;
+  sunrealtype* ydata = N_VGetArrayPointer(y);
+
+  a = SQRT(EVAL) + EVAL;
```

Comment:
To avoid the impression that user must use our macros for switching between things like `sqrt`, `sqrtf`, and `sqrtl` example have been gradually moving to defining their own macros for these purposes (in this case `SQRT`). This is defined in the shared examples header file (more comments on that below)